### DISCUSSION TITLE
Using Script I cannot get the temperature-sensor value

### DISCUSSION THREAD

**Discussion by Bebuccio** [arendst/Tasmota#18585] [Q&A] [answered]
I have an ESP32 compiled with script (and so without rules).
I want convert the old rule into script program.
The old rule was
rule1 on AM2301#Temperature>25.1 do power1 1 endon on AM2301#Temperature<25.1 do power1 0 endon

The script that I have written is:

>D
m:temp=0 ;temperature (avg -> temp[-2])
tavg=0   ;average temperatura 

>F
; every 100ms
temp=AM2301#Temperature

>S
; every second

; check every 10 secs
if upsecs%10==0 {
 print every 10 seconds temp = %AM2301#Temperature%

 ; test if temperature change
 tavg=temp[-2]
 if chg[tavg]>0 {
  print change temperature %tavg%
  =#acttemp
 }
}

; Function for temperature activity 
#acttemp
 if ((tavg>=25.1) and (pwr[1]<=0)) {
  ->Power1 1
 }
 if ((tavg<=25.1) and (pwr[1]>=1)) {
  ->Power1 0
 }

The output on the console is:
14:12:18.648 **var not found**: AM2301#Temperature
14:12:18.748 var not found: AM2301#Temperature
14:12:18.848 var not found: AM2301#Temperature
14:12:18.902 every 10 seconds temp = **???**
14:12:18.949 var not found: AM2301#Temperature
14:12:19.049 var not found: AM2301#Temperature

the problem is that the variable **AM2301#Temperature** is not recognized by script
What is the problem?

**Comment by gemu2015** [comment, marked as answer]
JSON sensor payloads may only be read in section >T
so you must copy the payload into a local variable 
also the moving average is done different.

```
>D
tmp=0
M:temp=0

>T
tmp=AM2301#Temperature

>S
; update moving average every second, so you get average over 8 seconds
temp=tmp

;when you read temp you always get the moving average of tmp;  (array[-2] is different don't use it here)

```
